ifupdown2/lib/iproute2.py

```diff
@@ -330,9 +332,9 @@ def get_vxlan_peers(self, dev, svcnodeip):
 
     ###
 
-    @staticmethod
-    def link_add_xfrm(ifname, xfrm_name, xfrm_id):
+    def link_add_xfrm(self, ifname, xfrm_name, xfrm_id):
         utils.exec_commandl(['ip', 'link', 'add', xfrm_name, 'type', 'xfrm', 'dev', ifname, 'if_id', xfrm_id])
+        self.__update_cache_after_link_creation(xfrm_name, None)
```

Comment:
`kind=None`

This is probably fine for now but might be needed in the future no? I'd prefer if we would set kind="xfrm" now to avoid problems in the future.

Reply:
I always got an exception when I did that but maybe I am running into the same problem as your fix for ip6ip6.
Let me try that out when I am back later today, if that fixes it I will change the patch to us ekind="xfrm"

Reply:
You are right, a similar fix is needed for xfrm. It would be great if you could include that in your update, thanks! :)

Reply:
Alright, I changed the code and tested it on my end.
It works fine now with xfrm as IFLA_INFO_KIND. 
Thanks for your help :)